Applications/LogisticRegression/jni/main.cpp

```diff
@@ -54,24 +67,110 @@ float stepFunction(float x) {
   return x;
 }
 
+/**
+ * @brief     get idth Data
+ * @param[in] F file stream
+ * @param[out] outVec feature data
+ * @param[out] outLabel label data
+ * @param[in] id id th
+ * @retval boolean true if there is no error
+ */
+bool getData(std::ifstream &F, std::vector<float> &outVec,
+             std::vector<float> &outLabel, unsigned int id) {
+  std::string temp;
+  F.clear();
+  F.seekg(0, std::ios_base::beg);
+  char c;
+  unsigned int i = 0;
+  while (F.get(c) && i < id)
+    if (c == '\n')
+      ++i;
+  F.putback(c);
+
+  std::getline(F, temp);
+  std::istringstream buffer(temp);
+  float x;
+  for (unsigned int j = 0; j < feature_size; ++j) {
+    buffer >> x;
+    outVec[j] = x;
+  }
+  buffer >> x;
+  outLabel[0] = x;
+
+  return true;
+}
+
+/**
+ * @brief     get Data as much as batch size
+ * @param[out] outVec feature data
+ * @param[out] outLabel label data
+ * @param[out] last end of data
+ * @param[in] user_data user data
+ * @retval int 0 if there is no error
+ */
+int getBatch_train(float **outVec, float **outLabel, bool *last,
+                   void *user_data) {
+  std::ifstream dataFile(data_file);
+  unsigned int data_size = total_train_data_size;
+  unsigned int count = 0;
+
+  if (data_size - train_count < batch_size) {
+    *last = true;
+    train_count = 0;
+    return 0;
+  }
+
+  for (unsigned int i = train_count; i < train_count + batch_size; ++i) {
+
+    std::vector<float> o;
+    std::vector<float> l;
+    o.resize(feature_size);
+    l.resize(1);
+
+    if (!getData(dataFile, o, l, i)) {
```

Comment:
Could you check if this function always returns true?

Reply:
fixed. Thanks.